src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1138,21 +1138,21 @@ export class SavedObjectsRepository {
     if (foundExactMatch && foundAliasMatch) {
       result = {
         // @ts-expect-error MultiGetHit._source is optional
-        saved_object: getSavedObjectFromSource(this._registry, type, id, exactMatchDoc),
+        savedObject: getSavedObjectFromSource(this._registry, type, id, exactMatchDoc),
```

Comment:
We're following snake_case convention for Core's HTTP API's. This bleeds into the source code naming convention, but we've decided to use snake case in our source code if it reduces the complexity of having a snake_case HTTP API (longer discussion here https://github.com/elastic/kibana/issues/52284)

Reply:
Great, so instead of changing the server interface to camelCase, I'll change the public interface to snake_case!

Reply:
Updated in 2e2fa20 and 1c95043 so both client and server interfaces use `{ saved_object, outcome, alias_target_id }` for the resolve response.